### PR TITLE
V2.4.10 particle transform fix

### PR DESCRIPTION
**Pull request by zxx43**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

https://forum.cocos.org/t/topic/133676

### CODE REVIEW

File: cocos2d/renderer/core/base-renderer.js [hunk 1]
```diff
@@ -302,6 +302,16 @@ export default class Base {
...
 
+    if (node.parent) {
+      node.parent.getWorldMatrix(_m4_tmp);
+    } else {
+      _m4_tmp.set(Mat4.IDENTITY);
+    }
+    device.setUniform('cc_matParent', Mat4.toArray(_float16_pool.add(), _m4_tmp));
+
+    node.getLocalMatrix(_m4_tmp);
+    device.setUniform('cc_matLocal', Mat4.toArray(_float16_pool.add(), _m4_tmp));
```
Comment (SantyWang) [inline cocos2d/renderer/core/base-renderer.js]: 不应该在渲染流程里面去设置吧，应该在粒子系统里面去设置呢？

Comment (zxx43) [reply cocos2d/renderer/core/base-renderer.js]: 粒子的updateMaterial只跑一次

Comment (jk20012001) [reply cocos2d/renderer/core/base-renderer.js]: 材质参不一定保证每帧都能设置到吧, 变换矩阵是需要每帧设置的, 如果改到粒子材质里设置会不会有问题

Comment (zxx43) [reply cocos2d/renderer/core/base-renderer.js]: 改了

Comment (SantyWang) [reply cocos2d/renderer/core/base-renderer.js]: > 材质参不一定保证每帧都能设置到吧, 变换矩阵是需要每帧设置的, 如果改到粒子材质里设置会不会有问题

这个地方是每帧都走的，而且 3.x 的粒子也是这么玩的

**Pull request closed without merging**